webserver/views/datasets.py

```diff
@@ -304,7 +305,7 @@ def edit(dataset_id):
 
 
 @datasets_bp.route("/<uuid:dataset_id>/delete", methods=("GET", "POST"))
-@login_required
+@api_login_required
```

Comment:
same as `import_csv` here, this page only returns HTML (even though there's a jsonify here it's a bug)